internal/pkg/pct/install_test.go

```diff
@@ -39,71 +56,234 @@ func TestInstall(t *testing.T) {
+				},
+			},
+			mocks: mocks{
+				dirs: []string{
+					templatePath,
+					extractionPath,
+					filepath.Join(extractionPath, "good-project"),
+				},
+				files: map[string]string{
+					filepath.Join(templatePath, "good-project.tar.gz"): string(tarballBytes),
+					filepath.Join(extractionPath, "good-project", "pct-config.yml"): `---
+template:
+  id: good-project
+  author: puppetlabs
+  version: 1.0.0
+`,
+				},
+			},
 		},
 		{
 			name: "if it fails to gunzip",
 			args: args{
 				templatePath: filepath.Join(templatePath, "good-project.tar.gz"),
 				targetDir:    templatePath,
 			},
-			gunzipErr:     true,
-			expectedError: "gunzip error",
+			mockReponses: mockReponses{
+				gunzip: []mock.GunzipResponse{
+					{
+						ErrResponse: true,
+					},
+				},
+			},
+			mocks: mocks{
+				dirs: []string{
+					templatePath,
+				},
+				files: map[string]string{
+					filepath.Join(templatePath, "good-project.tar.gz"): string(tarballBytes),
+				},
+			},
+			expected: expected{
+				errorMsg: "gunzip error",
+			},
 		},
 		{
 			name: "if it fails to untar",
 			args: args{
 				templatePath: filepath.Join(templatePath, "good-project.tar.gz"),
 				targetDir:    templatePath,
 			},
-			untarFail:     true,
-			expectedError: "untar error",
+			mocks: mocks{
+				dirs: []string{
+					templatePath,
+				},
+				files: map[string]string{
+					filepath.Join(templatePath, "good-project.tar.gz"): string(tarballBytes),
+				},
+			},
+			mockReponses: mockReponses{
+				gunzip: []mock.GunzipResponse{
+					{
+						Fail:     false,
+						FilePath: filepath.Join(extractionPath, "good-project.tar"),
+					},
+				},
+				untar: []mock.UntarResponse{
+					{
+						ErrResponse: true,
+					},
+				},
+			},
+			expected: expected{
+				errorMsg: "untar error",
+			},
 		},
+		{
+			name: "If the tarball doesnt contain an valid config",
+			args: args{
+				templatePath: filepath.Join(templatePath, "good-project.tar.gz"),
+				targetDir:    extractionPath,
+			},
+			expected: expected{
+				errorMsg: "Invalid config: open " + filepath.FromSlash("path/to/extract/to/good-project/pct-config.yml") + ": file does not exist",
+			},
+			mockReponses: mockReponses{
+				untar: []mock.UntarResponse{
+					{
+						ReturnPath:  filepath.Join(extractionPath, "good-project"),
+						ErrResponse: false,
+					},
+					{
+						ReturnPath:  filepath.Join(extractionPath, "puppetlabs", "good-project", "1.0.0"),
+						ErrResponse: false,
+					},
+				},
+				gunzip: []mock.GunzipResponse{
+					{
+						Fail:     false,
+						FilePath: filepath.Join(extractionPath, "good-project.tar"),
+					},
+				},
+			},
+			mocks: mocks{
+				dirs: []string{
+					templatePath,
+					extractionPath,
+					filepath.Join(extractionPath, "good-project"),
+				},
+				files: map[string]string{
+					filepath.Join(templatePath, "good-project.tar.gz"): string(tarballBytes),
+				},
+			},
+		},
+		// This test cannot be performed due to a bug in aferos MemMapFs
```

Comment:
Can this test be re-enabled given [your observation](https://github.com/spf13/afero/issues/319)?